src/core/ticker/Ticker.js

```diff
@@ -217,18 +221,86 @@ export default class Ticker
      *
      * @param {Function} fn - The listener function to be added for one update
      * @param {Function} [context] - The listener context
+     * @param {number} [priority=10] - The priority for emitting
+     * @returns {PIXI.ticker.Ticker} This instance of a ticker
+     */
+    addOnce(fn, context, priority = Ticker.PRIORITY_DEFAULT)
+    {
+        return this._add(fn, context, priority, true);
+    }
+
+    /**
+     * Internally adds the event handler so that it can be sorted by priority.
+     * Priority allows certain handler (user, AnimatedSprite, Interaction) to be run
+     * before the rendering.
+     *
+     * @private
+     * @param {Function} fn - The listener function to be added for one update
+     * @param {Function} context - The listener context
+     * @param {number} priority - The priority for emitting
+     * @param {boolean} once - If it should only be added once
      * @returns {PIXI.ticker.Ticker} This instance of a ticker
      */
-    addOnce(fn, context)
+    _add(fn, context, priority, once)
     {
-        this._emitter.once(TICK, fn, context);
+        const listeners = this._listeners;
+        const emitter = this._emitter;
+
+        listeners.push(new TickerListener(fn, context, priority, once));
+
+        // Sort by prority from highest to lowest
+        listeners.sort((listener1, listener2) => listener2.priority - listener1.priority);
```

Comment:
The copy might not be needed anymore, but this was my though process... consumers can modify the listeners array from any index? So during an emit, you might be dereferencing the member array, but it is still a reference, and can still be modified by the listeners invoked the emit method, that is if this method didn't copy up front. e.g. Add 2 listeners, have the first listener add a listener at higher priority than itself, and make that listener added this way do the same. What my theory is, is since you capture the `len` in the emit loop, which should be fine, because it's more deterministic this way, and because the 1st listener added another at higher priority (a), and that added another at yet even higher priority (b). The emit will invoke listener 1, listener 1. Listener 2, listener a, and listener b are not invoked. It's good that listener a and b are not, but listener 2 should have been.

If you can think of another way that this nondeterministic can be addressed, go for it. But your previous implementation of `sort` is a copy. So this does the same thing with a `slice`.